packages/mdc-segmented-button/segment/_segment-theme.scss

```diff
@@ -0,0 +1,90 @@
+//
+// Copyright 2020 Google Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Theme: Variables, functions, and mixins required to theme component segment
+
+@use "@material/feature-targeting/functions" as feature-targeting-functions;
+@use "@material/feature-targeting/mixins" as feature-targeting-mixins;
+@use "@material/theme/variables" as theme-variables;
+@use "@material/theme/custom-properties";
+
+$on-surface: theme-variables.prop-value(on-surface);
+$surface: theme-variables.prop-value(surface);
+$primary: theme-variables.prop-value(primary);
+
+$unselected-ink-color: custom-properties.create(
+  --mdc-segmented-button-unselected-ink-color,
+  rgba($on-surface, 0.6)
+);
+$unselected-outline-color: custom-properties.create(
+  --mdc-segmented-button-unselected-outline-color,
+  rgba($on-surface, 0.12)
+);
+$unselected-container-fill-color: custom-properties.create(
+  --mdc-segmented-button-unselected-container-fill-color,
+  rgba($surface, 1)
+);
+$selected-ink-color: custom-properties.create(
+  --mdc-segmented-button-selected-ink-color,
+  rgba($primary, 1)
+);
+$selected-outline-color: custom-properties.create(
+  --mdc-segmented-button-selected-outline-color,
+  rgba($primary, 1)
+);
+$selected-container-fill-color: custom-properties.create(
+  --mdc-segmented-button-selected-container-fill-color,
+  rgba($primary, 0.08)
+);
+
+$height: 36px;
+$min-width: 48px;
+$horizontal-padding: 12px;
+$border-width: 1px;
+$border-radius: 4px;
+$icon-width: 24px;
+$icon-font-size: 18px;
+$label-padding: 6px;
+
+@mixin ink-color($color, $query: feature-targeting-functions.all()) {
```

Comment:
I think you'll need a separate mixin for `selected-ink-color()` so that client do not have to worry using MDC selectors.

```scss
.custom-segmented-button {
  // This mixin emits appropriate selector (i.e., `__segment--selected`) required to set color to selected segment.
  @include selected-ink-color($foo-color);
}
```

Reply:
I see what you mean. Should I add the ability to do this to the other `selected` colors, or use a single mixin to modify all of them, since they all use `theme-variables.$primary`. Also, how much of the component should be themed? At this point I just have colors, but should all of the colors be customizable?

Reply:
Yes, all colors should be customizable (Especially state colors). It is best to break down the mixin so that it serves a single purpose.